compiler/modules/types/core.bal

```diff
@@ -778,6 +782,24 @@ public function isNever(SemType t) returns boolean {
     return t is BasicTypeBitSet && t == 0;
 }
 
+public function isCyclic(Context cx, SemType t) returns boolean {
```

Comment:
I don't like the implementation of this function. You're doing similar stuff is isMemberNever.

We need to factor out functions for trying to do conversions between a SemType to an single basic type/Bdd/atom.

Reply:
fixed